ambry-router/src/main/java/com/github/ambry/router/TtlUpdateOperation.java

```diff
@@ -388,6 +403,61 @@ private void checkAndMaybeComplete() {
     }
   }
 
+  /**
+   * Try to replicate the blob and then retry the TtlUpdate Request
+   * @return true if it's under retry. return false if cannot do retry or retry is finished.
+   */
+  private boolean retryWithReplicateBlob() {
+    // the conditions to trigger ReplicateBlob operation:
+    // 1. at least one replica returned successful status
+    // 2. and at least one replica returned NOT_FOUND status.
+    if (!operationTracker.hasNotFound() || operationTracker.getSuccessReplica().size() == 0) {
```

Comment:
double negatives are confusing
not hasNotFound == hasFound 

please make a helper hasFound()

Reply:
Also why return when the operationTracker.getSuccessReplica().size() == 0 ?
Doesn't it mean no replicas were found to have the blob ?

Reply:
Yes, as the comment, these two conditions are 
1. At least there is one replica returned NOT_FOUND. So it makes sense to do the ReplicateBlob.
2. At least here is one replica is successful. So we can get the Blob from that replica.

Change it to "hasFound" logically is not not we want to express.

Reply:
1. So the first part of your comment explains the first part of your boolean and the second part explains the second part of the boolean ?
2. A negation of AND expression instead of OR makes more sense - we want to replicate if at least one replica returned NOT_FOUND && another one returned success  otherwise we don't replicate ?

```
if (! (operationTracker.hasNotFound() && operationTracker.getSuccessReplica().size() > 0)) {
  return false
}
```